### ISSUE TITLE
[WARN] Riseup.net Email service don't delete your emails, keep even after you delete them!!

### ISSUE THREAD

**Issue by ghost** [nylira/prism-break#772]
PLEASE HIDE RISEUP FROM EMAIL SECTION
## https://user.riseup.net/ticket/615275

Ticket 615275
Subject Urgency     

If the backup option is unchecked and Thunderbird deletes emails after downloading them, are they stored on Riseup servers?
Is there any chance that a government agency could retrieve these emails after seizing a server, perhaps by scanning for deleted files?
- [x] taggart, August 16, 2013 00:01
  Hi,
  We don't keep deleted messages.
  All of our servers use full disk encryption, so an attacker would not be able to find them by scanning for deleted files either.
  Thanks,
  taggart
- [x] krutt, December 16, 2013 14:59
  I have just changed my Riseup user-name.
  After signing into the web-mail with the new user-name, I saw 162 messages that I thought had been deleted.
  I feel betrayed, as I was under the impression e-mails were not stored on the servers after being deleted.
  I never had the backup option enabled. http://i.imgur.com/yH2pdIQ.png
- [x] anoooonx, December 17, 2013 01:29
  Me too. I always purge(delete trashbin) emails after I read.
  I only use WebMail(Squirrel).
  Today when I login to webmail, I found that my old emails are still in the trashbin.
  I think I should move to A/I.
  http://www.autistici.org/en/index.html
  DO NOT LIE to customers, riseup.
- [x] taggart, December 18, 2013 00:50
  Hi,
  I think there is some confusion here. Many mail clients (including webmail clients) by default move mails to a 'Trash' folder when you mark them for deletion. Also complicating things is the fact that there can be a local 'Trash' folder stored on your computer and also an IMAP 'Trash' folder which is stored on the server, and the behavior is dependent on the settings in your mail client.
  For the IMAP 'Trash' folder stored on the server, Riseup automatically deletes messages older than 21 days. Riseup has no control over any local folders stored on your computer.
  Riseup also has no control over user mail client settings for local clients(Thunderbird, etc). For the webmail clients, new users get default settings and can adjust them from there. Now might be a good time for Riseup to consider adjusting the default mail settings for webmail. The current 'Trash' behavior has always been seen as a feature, but with recent changes it might be better to lose the feature in exchange for storing less email.
  We document information about these special folders on help.riseup.net, but I can see how it could still catch people off guard. I think it would be useful to add additional help pages describing how users could adjust their client settings to disable this 'Trash' feature to ensure that mail wasn't stored at all. Probably we need one page describing the problem and then each of the help pages for specific clients should have more instructions with screenshots showing how to turn it off for that specific client.
  Hopefully the above is clear. Let us know if you have any questions or comments.
  taggart
- [x] krutt, December 18, 2013 20:15
  I did a test using webmail only. Here's what I did:
  1) Changed username to zgggz
  2) Sent an email to myself (Subject: "Email not deleted proof")
  3) Checked inbox in webmail and saw the welcome email "Welcome to Riseup" AND the email I just wrote
  4) Clicked on Inbox folder in webmail, ticked both emails, clicked delete
  5) Clicked on Trash folder in Webmail, ticked both emails, clicked delete
  At this point my inbox size for all folders shows up as 0.0kB. I cannot find any emails anywhere.
  6) Changed username to zgggz1
  7) Checked inbox in webmail
  8) SAW THE EMAIL I HAD DELETED, AND THEN PURGED FROM THE TRASH TITLED "EMAIL NOT DELETED PROOF"
  9) Changed username back to krutt
  ???
- [x] anoooonx, December 20, 2013 09:54
  This is a major security risk.
  It seems that Riseup is telling us a lie, no offense here.
  I think I should make an account in "Autistici/Inventati".
  http://www.autistici.org/en/index.html
  http://www.infoaut.org/index.php/english/item/8937-freedom-and-rights
- [x] anoooonx, December 20, 2013 10:03
  0) Login to webmail, Delete ALL EMAILS. Make sure trash is empty.
  1) Change a username from user.riseup.net
  2) Logout and wait 5 minutes
  3) Login to webmail
  And... WHAT THE F...
  Every emails (ALL from I open an account TILL today) which I surely delete appear in INBOX!!!!! (WebMail/INBOX)
  Don't believe me? Then try for yourself!!
  Riseup is surely don't delete user's emails! Explanation needed!
  This is a news, I'm gonna tell this scandal to my friends!!
- [x] krutt, December 20, 2013 13:43
  anoooon:
  I'd like to add that in my case,
  I could also see every single e-mail from the day I signed up after going through these steps.
  However, clicking on "destroy mailbox", which completely deletes your email service,
  and making a new one seemed to delete (or just hide? I don't know) the emails.
- [x] anoooon, December 20, 2013 13:52
  |> clicking on "destroy mailbox"
  Uh, where? Would you tell me where that button exist?
  |> making a new one seemed to delete (or just hide? I don't know) the emails
  I think they just "hide" your mails. No deletion.
- [x] anoooon, December 20, 2013 13:56
  |> I could also see every single e-mail from the day I signed up
  And when do you join riseup?
  I join here in December 2013, if my memory was right.
- [x] anoooon, December 20, 2013 13:58
  krutt, IF,
  you are being an member more than 21 days,
  the information below is a goddamn LIE, isn't it? :)
  |> taggart, December 18, 2013 00:50
  |> ... For the IMAP 'Trash' folder stored on the server,
  |> Riseup automatically deletes messages older than 21 days.
- [x] krutt, December 20, 2013 14:31
  anoooon:
  |> Uh, where? Would you tell me where that button exist?
  If you visit the e-mail tab, you can see it in the bottom right corner.
  |> And when do you join riseup?
  I've been here for a few months, the earliest email that should have been deleted
  is dated towards the end of September on the screenshot in one of my previous posts.
  |> krutt, IF, you are being an member more than 21 days, the information below is a goddamn LIE, isn't it? :)
  taggart might not be aware of this issue, so I can't call him a liar.
  But the fact is, that messages which had been deleted for well over 21 days appeared in my inbox.
  I am keen on hearing an official reply on this.
  Anoooon, please don't jump to conclusions too fast.
  There may be a rational to all of this, although I am feeling less comfortable by the minute.
- [x] anoooon, December 20, 2013 14:40
  0) Email settings: Destroy mailbox. Destroy and create it again. [done](Keep backup is OFF, of course.)
  1) Send an email to myself, and delete it. Trash is empty. [done]
  2) I'll change my nick (now)
- [x] anoooon, December 20, 2013 14:54
  3) Login to Webmail as "anoooonxp"...
  Booo.... Negative. Email come back alive.
  (Those 2 things: I delete from a trashbox. Not these appears in INBOX!)
  Subject sort From sort Date sort Size sort
  This message is new Head down!! anoooon@riseup.net 6:41 am 1.3 k
  This message is new Welcome to Riseup! Riseup Collective 6:39 am 87 k
  |> making a new one seemed to delete (or just hide? I don't know) the emails.
  is not work. Riseup STILL don't delete emails.
  I leave these 2 things in INBOX. Just logout...
  4) change nick back to anoooon.
  5) Login to webmail.... and...
  huh!? Where is these 2 mails go!?
  Subject sort From sort Date sort Size sort
  ## |THIS FOLDER IS EMPTY
  
  Conclusion:
  Riseup is NOT deleting user's email.
  Emails which age is 21-days older still exist in riseup server.
  Disclaimer:
  Q1: You are using an email client. That's why your emails are exist in riseup!
  A1: Negative. I always use ONLY riseup's webmail(squirrel).
  Q2: You just have a dream. You don't delete an email.
  A2: Absolutely not. I always delete from INBOX/SENT and also delete from trash. I always leave no mails when I click logout.
  Q3: You are lying. Please don't kick us:( (by Riseup team)
  A3: I think you are a liars. Go try for yourself.
- [x] anoooon, December 20, 2013 14:57
  |>> This message is new Welcome to Riseup! Riseup Collective 6:39 am 87 k
  This mail was automatically send by riseup, so
  (Those "1"[not 2] things: I delete from a trashbox. Not these appears in INBOX!)
  |>> Head down!! anoooon@riseup.net 6:41 am 1.3 k
  This is an test mail which i sent to myself before changing an nick.
  I delete it as anoooon, change name to anoooonxp, and login to webmail.

**Comment by Muvuk** [comment]
I cant read linked page on riseup.net. How about this problem? I found a very similar thing. What do you think?
https://www.reddit.com/r/privacy/comments/2oa41g/fed_up_with_waiting_to_be_approved_by_autistici/cmlzqz5

**Comment by gagou9** [comment]
@Muvuk, the issue with riseup and not deleted emails has nothing to do with you link :)

Your link is about somebody complaining that Autistici takes some time to answer the account request, but this is *normal* behaviour! They are not a company, they don't work for money, they work to build better systems and also, they are humans, not machines. They have a lot of requests I'm sure, and they need time to answer everything. I got my autistici account in a few days or a week or so.

riseup and autistici are both trustworthy, and powered by human lives, by people who want to offer alternatives, so they are not going to do things in a profitable way.

**Comment by gagou9** [comment]
By the way the ticket is visible if you log into user.riseup.net.

**Comment by taggart** [comment]
The ticket https://user.riseup.net/ticket/615275 is only accessable to logged in riseup users. A lot of it was pasted into this thread above, but only up to Dec 20. There was some more confusing back and forth and delays, but here is the important part:

>taggart, January 13, 2014 14:45
>
>Hi,
>
>Thanks for being patient, the holidays and then sickness have meant we couldn't reply to this problem until now.
>
>There are a couple things going on here:
>
>1) A bug with our scripts for renaming accounts that didn't account for the new performance caches used by the imap server (dovecot indexes). This meant that, for a brief period of time, when you renamed an account, the caches were not properly purged which resulted in old mail showing up. It appears this is what happened in your case. This was clearly a bug, and thanks for pointing this out, because we were not aware of the problem! We did fix this and have confirmed the test you provided no longer causes old mail to show up, but it would be very helpful if you could also confirm it yourself.
>
>2) Despite this issue with the caches, my reply to the original question is still true:
>Q) Is there any chance that a government agency could retrieve these emails after seizing a server, perhaps by scanning for deleted files?
>A) All of our servers use full disk encryption, so an attacker would not be able to find them by scanning for deleted files either.
>
>However... if the attacker is somehow able to get past the encryption, then they would have access to the unencrypted filesystem and then they could scan for deleted files, dovecot indexes, and possibly other bugs that result in extra copies on the filesystem. We try our best, but we're not perfect.
>
>The only real solution to these problems is end-to-end encryption where only the client has the secret key. In the past this has meant using GPG, but LEAP (https://leap.se) are working on a better solution that Riseup plans to adopt when it's ready.
>As always, our preference is to store as little data for our users as possible---if you can use GPG until we have something better, that's great.
>
>taggart

The ticket has been resolved and we're not aware of any other issues, but we're always happy to discuss any concerns.

**Comment by thomhastings** [comment]
Whoa, did I read that right?
*won't* delte your e-mails?
Sorry anarchists, but the
right-to-be-forgotten is
a really important idea.
As is amnesia, as well
as ephemeral messenging.

Be well.

------------------------------------------------------
​-​
Re:
*​​member*
​_hack
history:
*​​Phrack#42:13*

*Preserve your right to privacy.*

*When replying to this email, I encourage*
*you to make use of PGP compliant encryption*​

2015-02-04 13:33 GMT-06:00 taggart <notifications@github.com>:

**Comment by Zegnat** [comment]
@thomhastings, you did not read that right. A bug in a caching mechanism made it impossible for the software to forget, even though the Riseup team believed the software to purge its memory.

Loads and loads of miscellaneous problems arise from caching, just think how often you would have to do a ‘full refresh’ of a website before a design or image can show up correctly all because your browser didn’t handle a specific edge-case. The same happened here, it was reported, and dealt with.

Thanks @taggart for reposting your definitive answer here.

As a side note, **this issue is closed**. If anyone has reason to believe Autistici or Riseup are no longer fit for PRISM Break please open a new issue dedicated to the suspected problem. Always remember that these are email providers chosen for how they have profiled themselves to be standing up for their user’s rights. Sign-up is hard at these services for a reason, over at A/I doubly so because of their relatively heavy (political) manifest, and does not seem to be a ground for removal from PRISM Break in my opinion.

**Comment by Muvuk** [comment]
@gagou9 

I read the next on that page, i think it is very similar (non deleted) problem like risup. That's the reason why i ask about in this thread. 

"I don't trust Autistici because - I have a few Autistic accounts. One had sent/received only two e-mails. One sent/received 5, one sent/received 23, one sent/received 136, one sent/received 621. Subsequently I deleted ALL e-mails and emptied trash. So all my accounts have no e-mails in them, including my trash can. However the amount of space used by each of my e-mail accounts is proportional to the number of e-mails I had sent/received. That means Autistici keeps my deleted e-mails on their servers, which is not cool, because I have already deleted them permanently from my mailbox. Why are they keeping my permanently deleted emails on their servers - I don't see any healthy explanation for that."

**Comment by piger** [comment]
In my experience with email at Autistici/Inventati I've found that there is always a rather long delay (about 6 o 7 days) between the deletion of emails from my IMAP client and the update of the statistics showed in the User Panel; I've asked on IRC for clarification and they told me those counters are effectivly updated only once a week.

**Comment by thomhastings** [comment]
Hi,

I&#39;d like to add you to my professional network on LinkedIn.

- Thom

Accept: https://www.linkedin.com/blink?simpleRedirect=3oRc3ASd3kTcj4Ve34OcP0TejkZh4BKrSBQonhFtCVF9zgNr65AinBHfnBBiShBsC5EsOpQsSlRpRZBt6BSrCAZqSkConhzbmlQqnpKqiRQsSlRpORIrmkZpSVFqSdxsDgCpnhFtCV9pSlipn9Mfm4Cc3cJujlDtj1UdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=


You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/blink?simpleRedirect=30PbnARpTkMu3lFbjBRoSBzcOQZp6BB9zBalD9MrBx4q7dihSNypztEskVgtQgJcBEUpk55mkx5qR1ViSVIr31ArCpdiCtRkA55lkJkrRlWunFAd6JikD5ElA8QuC99iT97j5wRdz5Bkmpys71fp35qjmVTrRFah45OoQtMqClAi38UsT0Tl6V8oRlggDhMhkB9cSoOcPwOszpPckZ7tjBTjld9rQRFcntnbl1QoQRFr4JHoSp6r6tNtkJ2ml1Im6pTkz1RlR1psDhlgk51gmJnmklxukNOrAZ8kk4Zp6BLr2pQsSlRpRZBt6BSrCAZqSkCkjoPp4l7q5p6sCR6kk4ZrClHrRhAqmQCrDlIfngCc3cJujlDtj1UdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&amp;msgID=I5970321922112651264_500&amp;markAsRead= Learn why we included this at the following link: https://www.linkedin.com/blink?simpleRedirect=0Ue3sQfmh9pmNzqnhOoipQsSlRpRZBt6BSrCAZqSkCr79lpmdFtD9BkT9BrmZQsTlzfm4Cc3cJujlDtj1UdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=
&copy; 2014, LinkedIn Corporation. 2029 Stierlin Ct. Mountain View, CA 94043, USA

**Comment by thomhastings** [comment]
Thom Hastings would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/blink?simpleRedirect=3oRc3ASd3kTcj4Ve34OcP0TejkZh4BKrSBQonhFtCVF9zgNr65AinBHfnBBiShBsC5EsOoVclZMu6lvtCVFfmJB9D9Bp6VFrmlObnhMpmdzoiRybmtSrCBvrmRLoORIrmkZpSVFqSdxsDgCpnhFtCV9pSlipn9Mfm4CtPcJrPhRsj9WdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=

View Thom Hastings&#39;s profile: https://www.linkedin.com/blink?simpleRedirect=0VclZMu6lvtCVFfmJB9D9Bp6VFrmlObnhMpmdzoiRybmtSrCBvrmRLoORIrmkZpSVFqSdxsDgCtD1KfngCtPcJrPhRsj9WdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=

You are receiving Reminder emails for pending invitations. Unsubscribe here: https://www.linkedin.com/blink?simpleRedirect=6RLoQkO9m9Rq7hFpQkO9nBIs6lOc3gBoCoPejsTcj1BoPAScm4OejdCdjdxpj0Ncj0Mc30Mc31CoP8VomcOp3wVp64SdPkUoj0McmkOem9ApzcQdPcMdShAdC4NcPtyomgNczwOdP8Nc30Mgz8BumNMpn8Zr6BxrmkCc3oMc38Zp6ACdzkMejoQdjsNcjAUcj8Pc3sVdjRAqmsCej5vs7xBnTpKqjRHpipytndKtiRCbmdypmRvrmRLoORIrmkZpSVFqSdxsDgCrSZDfngCtPcJrPhRsj9WdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=

You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/blink?simpleRedirect=tPcJrPhRsj9WdmAJenlzqmcPbjRAqmkCq4ANtnlAgRxgdkwUmAhptD1xmyR1l6cPqmlBpB1RhmJ4kOQPl7BbqPhQr3pQcmJeoltBgnFMrmpKoSJmdmdziQB6u4RLqD1ce5hysn1kbmxEsR0JcCZmtRpTbllNqCl9s49PskxDpPgJj4VecSh4oSFgc5lyinxpryQNhA9RtmB4ikJqpCpdqBh1nSAMgzxPiBt7oQJQtPlgu6tDjPcJh6hWm6lms7hitCFObnBUsnxVt5l1gk5TiSt4knlLizt1k4lhgjRAqmZI9zANnT1UplZSrCAZqSkCkjoPp4l7q5p6sCR6kk4ZrClHrRhAqmQCrDlIfngCtPcJrPhRsj9WdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&amp;msgID=I5970321922112651264_500&amp;markAsRead= Learn why we included this at the following link: https://www.linkedin.com/blink?simpleRedirect=3wUdPgZp4BBr6dFt79x9zANnT1UplZSrCAZqSkCr79lpmdFtD9BkT9BrmZQsTlzfm4CtPcJrPhRsj9WdmAJenlzqmcPbjRBfP9SbSkLrmZzbCVFp6lHrCBIbDtTtOYLeDdMt7hE&msgID=I5970321922112651264_500&markAsRead=
&copy; 2014, LinkedIn Corporation. 2029 Stierlin Ct. Mountain View, CA 94043, USA
